exonum/src/node/connect_list.rs

```diff
@@ -0,0 +1,197 @@
+// Copyright 2018 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Module that provides methods for managing the mapping between peers
+//! public keys and IP-addresses.
+
+use std::collections::BTreeMap;
+use std::net::SocketAddr;
+
+use blockchain::ValidatorKeys;
+use crypto::PublicKey;
+use helpers::fabric::NodePublicConfig;
+use node::ConnectInfo;
+
+// TODO: Don't reload whitelisted_peers if path the same. (ECR-172)
+
+/// `ConnectList` stores mapping between IP-addresses and public keys.
+#[derive(Debug, Clone, Serialize, Deserialize, Default)]
+pub struct ConnectList {
+    /// Peers to which we can connect.
+    #[serde(default)]
+    pub peers: BTreeMap<PublicKey, SocketAddr>,
+}
+
+impl ConnectList {
+    /// Returns `true` if a peer with the given public key can connect.
+    pub fn allow(&self, peer: &PublicKey) -> bool {
```

Comment:
I think this method should be named `peer_allowed`, like `address_allowed`

Reply:
Maybe even `is_peer_allowed`.

Reply:
In this case, `address_allowed` should also be changed